src/download.rs

```diff
@@ -80,12 +80,46 @@ impl Context {
         let placeholder = Message::load_from_db(self, placeholder_msg_id).await?;
         self.sql
             .transaction(move |transaction| {
-                transaction
-                    .execute("DELETE FROM msgs WHERE id=?;", paramsv![placeholder_msg_id])?;
+                // Move all the data from full message to placeholder.
+                // `id` stays the same, so foreign key constraints are not violated.
+                // For example, `reactions.msg_id` foreign key keeps pointing
+                // to the same message.
                 transaction.execute(
-                    "UPDATE msgs SET id=? WHERE id=?",
+                    "UPDATE msgs
+                     SET
+                     rfc724_mid=full.rfc724_mid,
+                     chat_id=full.chat_id,
+                     from_id=full.from_id,
+                     to_id=full.to_id,
+                     timestamp=full.timestamp,
+                     type=full.type,
+                     state=full.state,
+                     msgrmsg=full.msgrmsg,
+                     bytes=full.bytes,
+                     txt=full.txt,
+                     txt_raw=full.txt_raw,
+                     param=full.param,
+                     starred=full.starred,
+                     timestamp_sent=full.timestamp_sent,
+                     timestamp_rcvd=full.timestamp_rcvd,
+                     hidden=full.hidden,
+                     mime_headers=full.mime_headers,
+                     mime_in_reply_to=full.mime_in_reply_to,
+                     mime_references=full.mime_references,
+                     move_state=full.move_state,
+                     location_id=full.location_id,
+                     error=full.error,
+                     ephemeral_timer=full.ephemeral_timer,
+                     ephemeral_timestamp=full.ephemeral_timestamp,
+                     mime_modified=full.mime_modified,
+                     subject=full.subject,
+                     download_state=full.download_state,
+                     hop_info=full.hop_info
+                     FROM msgs AS full
+                     WHERE msgs.id=?1 AND full.id=?2",
```

Comment:
yip, this is better wrt constraints.

however, missing fields can become a hard to find source of bugs in the future (if we add a field to `msgs` by some migration and forget to list it here).

- is there an option to copy all but `id`?
- or maybe we can write a test that iterates over all fields, inserts unique values, calls `merge_messages()` and checks if all values are still the same.
- at a minimum, maybe add a comment after the last migration in `migrations.rs`

Reply:
> This test will also require maintenance, it has to updated each time a column is added.

we can maybe use `PRAGMA table_info(msgs);` for the test, i'll try that out and do a pr if it works out.